File: app/src/main/java/com/github/oryanmat/trellowidget/util/color/ColorUtil.java
package com.github.oryanmat.trellowidget.util.color;

import android.graphics.Color;

public class ColorUtil {
    public static int dim(int color) {
        float[] hsv = new float[3];
        Color.colorToHSV(color, hsv);
        // half the value component
        hsv[2] *= 0.5f;
        return Color.HSVToColor(hsv);
    }
}

File: app/src/main/kotlin/com/github/oryanmat/trellowidget/util/color/ColorUtil.kt
package com.github.oryanmat.trellowidget.util.color

import android.graphics.Color
import android.support.annotation.ColorInt

@ColorInt fun Int.dim(value: Float = .5f): Int {
    val hsv = FloatArray(3)
    Color.colorToHSV(this, hsv)
    // set the value component
    hsv[2] *= value
    return Color.HSVToColor(Color.alpha(this), hsv)
}

@ColorInt fun Int.lightDim(): Int = dim(.75f)